### PR TITLE
Documentation improvement, usage examples, checking return values in unirecfilter.c + Anonymizer.c bugfix

### LINKED ISSUE #22: Improvement of documentation of unirecfilter
README of UnirecFilter tells no information about PROTOCOL field.
This should be documented.

In addition, deeper analysis of this module would be helpful. The goal is to improve and complete docs for this module and add examples.

### PR DESCRIPTION
**Pull request by jaroslavh**
Closes #22 
- Added better documentation of usage of this module to README.md with examples
- Added code security checks throughout unirecfilter.c
- Documented function used in unirecfilter.c to unirecfilter.h

### CODE REVIEW

**Comment by jaroslavh** [comment]
Added fix to anonymizer.c according where I fixed 2 errors from Coverity Scan.

- fgets usage
- regexec usage

File: anonymizer/anonymizer.c [hunk 1]
```diff
@@ -212,7 +212,7 @@ void ip_anonymize(void *field_ptr, uint8_t mode)
 char *string_anonymize(void *field_ptr, uint32_t field_len, uint8_t mode, regex_t regex_IPV4, regex_t regex_IPV6)
 {
    int reti;
-   regmatch_t ip;
+   regmatch_t ip[2]; // expecting MAX 1 occurence, see man regexec for more info
```
Comment (cejkato2) [inline anonymizer/anonymizer.c]: `2` should be rather a (cpp macro) constant defined on one place - it improves readability and makes it easier to change this value just on one place in the future... on L225 and L227 there is the same constant with the same meaning.

Comment (cejkato2) [reply anonymizer/anonymizer.c]: Additionally, why 2? Why not just 1? (as it was...)

File: anonymizer/anonymizer.c [hunk 2]
```diff
@@ -364,7 +364,7 @@ int main(int argc, char **argv)
          fields_desc_len += strlen(anon_field_names[i]) + 2; // +2 for delimiter (", ")
       }
       fields_desc_len -= 2; // last delimiter
-      char* fields_desc = malloc(fields_desc_len + 1);
+      char *fields_desc = (char *) calloc(fields_desc_len + 1, sizeof(char));
```
Comment (cejkato2) [inline anonymizer/anonymizer.c]: Thanks for fixing the code according to the coding style.

File: unirecfilter/README.md [hunk 3]
```diff
@@ -64,15 +66,33 @@ Almost all data types from unirec are supported:
...
 
 #### File
-Filter specified in a file provides more flexibility. Format of the file is `[TEMPLATE_1]:FILTER_1;...;[TEMPLATE_N]:FILTER_N;` where each semicolon separated item corresponds with one output interface. One-line comments starting with `#` are allowed. To reload filter while unirecfilter is running, send signal SIGUSR1 (10) to the process.
+Filter specified in a file provides more flexibility. Format of the file is `TEMPLATE_1:FILTER_1;...;TEMPLATE_N:FILTER_N;` where each semicolon separated item corresponds with one output interface. One-line comments starting with `#` are allowed. To reload filter while unirecfilter is running, send signal SIGUSR1 (10) to the process.
 
 ## Default values
-You can use syntax FIELD=value in the template to specify default value used if field is not present on the input (f.e. uint32 BAR=1)
+You can use syntax FIELD=value in the template to specify default value used if field is not present on the input (f.e. uint32 BAR=1). Example bellow.
+`./unirecfilter -i u:test_ifc_in,u:test_ifc_out -O "ipaddr SRC_IP, ipaddr DST_IP, uint16 SRC_PORT, uint16 DST_PORT, string MESSAGE=\"this_is_message\""`
```
Comment (cejkato2) [inline unirecfilter/README.md]: Are you sure it is possible to put spaces into -O?

File: unirecfilter/unirecfilter.c [hunk 4]
```diff
@@ -153,22 +139,29 @@ void set_default_values(struct unirec_output_t *output_specifier) {
          // Trim quotes from string
          if (*(ptr+1) == '"') {
             ptr++;
-            *(skip_str_chr(ptr+1, '"')) = '\0';
+            *(skip_str_chr(ptr + 1, '"')) = '\0';
+         }
+         if (ur_set_from_string(output_specifier->out_tmplt, output_specifier->out_rec, ur_get_id_by_name(token), ptr + 1) != 0) {
+            fprintf(stderr, "set_defualt_values: Failed to get string from field.\n");
```
Comment (cejkato2) [inline unirecfilter/unirecfilter.c]: a typo: "defualt" instead of "default"

File: unirecfilter/unirecfilter.c [hunk 5]
```diff
@@ -218,19 +215,17 @@ int parse_file(char *str, struct unirec_output_t **output_specifiers, int n_outp
       // Beginning of filter
       case ':':
          beg_ptr = end_ptr + 1;
-         if ((end_ptr = skip_str_chr(end_ptr, ';')) == NULL) {
+                 if ((end_ptr = skip_str_chr(end_ptr, ';')) == NULL) {
```
Comment (cejkato2) [inline unirecfilter/unirecfilter.c]: This is some kind of "strange" indentation, isn't it?

File: unirecfilter/README.md (hunk 3 above)
Comment (jaroslavh) [reply unirecfilter/README.md]: Yeah, I have tested each command I have added to this README.

**Pull request merged**